tests/container-copy

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+set -eu
+
+# Install and setup LXD.
+install_lxd
+lxd init --auto
+
+cp /root/go/bin/lxd /var/snap/lxd/common/lxd.debug
+systemctl reload snap.lxd.daemon
+lxd waitready --timeout 100
+
+# Create a loop device that we can mount as a source for a dir storage pool.
+test_dir="$(mktemp -d XXX.test)"
+loopimg="$(mktemp "${test_dir}/XXX.img")"
+truncate -s 5G "${loopimg}"
+loopdev="$(losetup --show --find "${loopimg}")"
+mkfs.xfs "${loopdev}"
+disk_dir="$(mktemp -d "${test_dir}/XXX.disk")"
+mount "${loopdev}" "${disk_dir}"
+
+# Begin Test.
+set -x
+
+# Create a storage pool with the mounted disk.
+lxc storage create dir dir source="${PWD}/${disk_dir}"
+
+# Create a container which we will copy to another target.
+lxc launch "${TEST_IMG:-ubuntu-minimal-daily:22.04}" c1 --storage dir
+
+# Launch a VM which will be the target of the copy.
+lxc launch "${TEST_IMG:-ubuntu-minimal-daily:22.04}" target --vm
```

Comment:
Does it have to be a VM?

Reply:
Trying this with a container causes a bunch of permission denied errors. I'm not sure if that's expected or not, as it happens with non-externally mounted storage as well:

```
Error: Error transferring instance data: Failed migration on target: Failed creating instance on target: Rsync receive failed: /var/snap/lxd/common/lxd/storage-pools/default/containers/c1/: [exit status 23] (rsync: [generator] mknod "/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/dev/console" failed: Operation not permitted (1)
rsync: [generator] mknod "/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/dev/full" failed: Operation not permitted (1)
rsync: [generator] mknod "/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/dev/null" failed: Operation not permitted (1)
rsync: [generator] mknod "/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/dev/ptmx" failed: Operation not permitted (1)
rsync: [generator] mknod "/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/dev/random" failed: Operation not permitted (1)
rsync: [generator] mknod "/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/dev/tty" failed: Operation not permitted (1)
rsync: [generator] mknod "/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/dev/urandom" failed: Operation not permitted (1)
rsync: [generator] mknod "/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/dev/zero" failed: Operation not permitted (1)
rsync: [receiver] rsync_xal_set: lsetxattr("/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/var/log/journal","trusted.SGI_ACL_DEFAULT") failed: Operation not permitted (1)
rsync: [receiver] rsync_xal_set: lsetxattr("/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/var/log/journal","trusted.SGI_ACL_FILE") failed: Operation not permitted (1)
rsync: [receiver] rsync_xal_set: lsetxattr("/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/var/log/journal/5c9941407f4e46c5b6c652644b1c31b2","trusted.SGI_ACL_DEFAULT") failed: Operation not permitted (1)
rsync: [receiver] rsync_xal_set: lsetxattr("/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/var/log/journal/5c9941407f4e46c5b6c652644b1c31b2","trusted.SGI_ACL_FILE") failed: Operation not permitted (1)
rsync: [receiver] rsync_xal_set: lsetxattr("/var/snap/lxd/common/lxd/storage-pools/default/containers/c1/rootfs/var/log/journal/5c9941407f4e46c5b6c652644b1c31b2/.system.journal.wEww52","trusted.SGI_ACL_FILE") failed: Operation not permitted (1)
```
I also tried with `security.nesting=true` but that didn't change anything either.

Reply:
@tomponline Any thoughts about this one?

Reply:
Those errors sound as if the FS was mounted with `nodev` and without XATTR support, weird.

Reply:
I've removed the non-`mknod` errors as those were an issue with the build of LXD I was using for the test. On `latest/edge` the only errors are the `mknod` ones.

Reply:
Unprivileged containers cannot mknod.
You may be able to workaround this using a privileged container, e.g using some or all of this.

```
  raw.lxc: |-
    lxc.cgroup2.devices.allow = b 259:* rw
    lxc.cgroup.devices.allow = b 259:* rw
  security.nesting: "true"
  security.privileged: "true"
  ```
  
  We use this for running lxd-imagebuilder inside a container.

Reply:
Or potentially using security.syscalls.intercept.mknod see https://documentation.ubuntu.com/lxd/en/latest/syscall-interception/#available-system-calls

Reply:
That `security.syscalls.intercept.mknod` config sounds nice indeed however, my initial question of "why a VM" was because I feared the test would be slow. Turns out it's quite fast (4-5 minutes) so my concern was ill-founded.

Now if that `mknod` config works, it would be even faster as it would save downloading the 2nd image (for the VM) as it could use the same image twice.